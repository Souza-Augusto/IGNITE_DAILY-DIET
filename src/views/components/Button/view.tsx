import { ReactNode } from 'react';

import { TouchableOpacityProps } from 'react-native';
import { ButtonTypeStyleProps, Container, Title } from './styles';

type Props = TouchableOpacityProps & {
  title: string;
  before?: ReactNode;
  type?: ButtonTypeStyleProps;
};

export function Button({
  before,
  type = 'PRIMARY',
  title,
  style,
  ...rest
}: Props) {
  return (
    <Container type={type} style={style} activeOpacity={0.5} {...rest}>
      {before}
      <Title type={type}>{title}</Title>
    </Container>
  );
}
