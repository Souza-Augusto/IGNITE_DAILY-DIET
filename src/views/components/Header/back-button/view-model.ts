import { useNavigation } from '@react-navigation/native';

interface BackButtonProps {
  handleGoBack: () => void;
}

function useBackButtonViewModel(): BackButtonProps {
  const { goBack } = useNavigation();

  function handleGoBack() {
    goBack();
  }

  return { handleGoBack };
}

export { useBackButtonViewModel };
