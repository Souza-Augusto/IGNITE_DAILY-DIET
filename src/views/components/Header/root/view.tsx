import { ReactNode } from 'react';
import { Container } from './styles';
import { useNavigation } from '@react-navigation/native';

type Props = {
  leftElement?: ReactNode;
  rightElement?: ReactNode;
  children?: ReactNode;
  backButton?: boolean;
};

export function Root({ leftElement, rightElement, children }: Props) {
  return (
    <Container>
      {leftElement}
      {children}
      {rightElement}
    </Container>
  );
}
