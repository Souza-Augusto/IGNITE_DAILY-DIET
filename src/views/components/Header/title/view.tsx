import { Container, HeaderTitle } from './styles';

interface Props {
  title: string;
}

export function Title({ title }: Props) {
  return (
    <Container>
      <HeaderTitle>{title}</HeaderTitle>
    </Container>
  );
}
