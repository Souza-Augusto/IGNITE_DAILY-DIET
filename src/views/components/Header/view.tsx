import { Container, Logo, Profile } from './styles';
import { useNavigation } from '@react-navigation/native';

type Props = {
  showBackButton?: boolean;
};

export function Header({ showBackButton = false }: Props) {
  const navigation = useNavigation();

  return (
    <Container>
      <Logo source={require('@assets/images/png/Logo.png')} />
      <Profile source={require('@assets/images/png/Ellipse.png')} />
    </Container>
  );
}
