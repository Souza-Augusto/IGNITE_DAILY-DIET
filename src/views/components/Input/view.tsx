import { Container } from './styles';
import { TextInputProps } from 'react-native';
import theme from '@theme/index';

export function Input({ ...rest }: TextInputProps) {
  return (
    <Container
      selectionColor={theme.COLORS.GRAY_500}
      placeholderTextColor={theme.COLORS.GRAY_400}
      {...rest}
    />
  );
}
