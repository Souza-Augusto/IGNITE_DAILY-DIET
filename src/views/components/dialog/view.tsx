import { Modal, ModalProps } from 'react-native';
import {
  ConfirmButton,
  DialogContainer,
  TitleAndButtonContainer,
  DialogTitle,
  DialogButtonsContainer,
  CancelButton,
  CancelButtonTitle,
  ConfirmButtonTitle,
  TranparentButton,
} from './styles';

export interface Props extends ModalProps {
  dialogVisible: boolean;
  dialogMessage: string;
  positiveButtonTitle?: string;
  negativeButtonTitle?: string;
  negativeFunction?: () => void;
  positiveFunction?: () => void;
}

export function Dialog({
  dialogVisible,
  dialogMessage,
  negativeButtonTitle,
  positiveButtonTitle,
  negativeFunction,
  positiveFunction,
  ...rest
}: Props) {
  return (
    <Modal visible={dialogVisible} transparent {...rest}>
      <DialogContainer>
        <TranparentButton onPress={negativeFunction} />
        <TitleAndButtonContainer>
          <DialogTitle>{dialogMessage}</DialogTitle>
          <DialogButtonsContainer>
            {negativeButtonTitle && (
              <CancelButton onPress={negativeFunction}>
                <CancelButtonTitle numberOfLines={1}>
                  {negativeButtonTitle}
                </CancelButtonTitle>
              </CancelButton>
            )}
            {positiveButtonTitle && (
              <ConfirmButton onPress={positiveFunction}>
                <ConfirmButtonTitle>{positiveButtonTitle}</ConfirmButtonTitle>
              </ConfirmButton>
            )}
          </DialogButtonsContainer>
        </TitleAndButtonContainer>
        <TranparentButton onPress={negativeFunction} />
      </DialogContainer>
    </Modal>
  );
}
