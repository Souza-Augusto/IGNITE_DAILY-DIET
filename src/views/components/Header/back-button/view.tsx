import { BackIconContainer, BackIcon } from './styles';
import { useBackButtonViewModel } from './view-model';

interface Props {
  color?: string;
}

export function BackButton({ color }: Props) {
  const { handleGoBack } = useBackButtonViewModel();

  return (
    <BackIconContainer testID='back-button' onPress={handleGoBack}>
      <BackIcon name='arrowleft' color={color} />
    </BackIconContainer>
  );
}
