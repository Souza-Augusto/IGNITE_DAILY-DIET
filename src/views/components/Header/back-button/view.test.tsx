import { useBackButtonViewModel, BackButtonProps } from './view-model';
import { BackButton } from './view';
import { fireEvent, render } from '@__tests__/utils/customRenderItem';

beforeEach(() => jest.clearAllMocks());

jest.mock('./view-model');
jest.mock('@expo/vector-icons');

const mockedGoback = jest.fn();
const createMockHeaderBackButtonViewModel = (): BackButtonProps => ({
  handleGoBack: mockedGoback,
});

function makeSut() {
  const mockHeaderBackButtonViewModel = createMockHeaderBackButtonViewModel();
  (useBackButtonViewModel as jest.Mock).mockReturnValueOnce(
    mockHeaderBackButtonViewModel
  );

  const sut = render(<BackButton />);

  return sut;
}

describe('Component:back-button view', () => {
  it('should be render correctly', () => {
    const { toJSON, getByTestId } = makeSut();

    const container = getByTestId('back-button');
    expect(container).toBeDefined();
    expect(toJSON()).toMatchSnapshot();
  });
  it('should call handleGoback function', () => {
    const { toJSON, getByTestId } = makeSut();

    const container = getByTestId('back-button');
    fireEvent.press(container);

    expect(mockedGoback).toHaveBeenCalledTimes(1);
    expect(toJSON()).toMatchSnapshot();
  });
});
