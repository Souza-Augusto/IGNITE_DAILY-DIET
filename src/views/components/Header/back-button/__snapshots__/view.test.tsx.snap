// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component:back-button view should be render correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="back-button"
/>
`;

exports[`Component:back-button view should call handleGoback function 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="back-button"
/>
`;
